### PR TITLE
Implement correct mapping for contract creation code

### LINKED ISSUE #446: get_sourcecode_info in contract needs to do this for the contract creation to after recent extension to contract creation analysis
## Description

Lookup of sourcecode info may fail or return the wrong result when an issue was found in the constructor code. To Fix this the srcmap in addition to srcmap-runtime has to be added to the contract and the lookup then needs information on whether or not to look in the creation or transactional mapping in addition to the address.

## How to Reproduce

Contract: minmal.sol

```console
pragma solidity ^0.4.24;

contract Minimal {

	constructor(uint8 var1){
		assert(var1>0);
	}
}
```

```console
$ myth -x minimal.sol
Traceback (most recent call last):
  File "../../mythril/myth", line 9, in <module>
    mythril.interfaces.cli.main()
  File "/home/eltano/mythril/mythril/interfaces/cli.py", line 209, in main
    max_depth=args.max_depth, execution_timeout=args.execution_timeout)
  File "/home/eltano/mythril/mythril/mythril.py", line 385, in fire_lasers
    issue.add_code_info(contract)
  File "/home/eltano/mythril/mythril/analysis/report.py", line 38, in add_code_info
    codeinfo = contract.get_source_info(self.address)
  File "/home/eltano/mythril/mythril/ether/soliditycontract.py", line 105, in get_source_info
    solidity_file = self.solidity_files[self.mappings[index].solidity_file_idx]
TypeError: list indices must be integers or slices, not NoneType
...
$
```

## Expected behavior

Return an issue of an assert exception code that can be reached.

### PR DESCRIPTION
**Pull request by norhh**
fixes #446 , #473

### CODE REVIEW

File: mythril/ether/ethcontract.py [hunk 1]
```diff
@@ -17,7 +17,8 @@ def __init__(self, code, creation_code="", name="Unknown", enable_online_lookup=
         code = re.sub(r'(_+.*_+)', 'aa' * 20, code)
 
         self.code = code
-        self.disassembly = Disassembly(self.code, enable_online_lookup=enable_online_lookup)
+        self.disassembly = Disassembly(code, enable_online_lookup=enable_online_lookup)
+        self.creation_disassemble = Disassembly(creation_code, enable_online_lookup=enable_online_lookup)
```
Comment (JoranHonig) [inline mythril/ether/ethcontract.py]: disassembly

Comment (JoranHonig) [inline mythril/ether/ethcontract.py]: An eth contract is used to model contracts that we don't have the construction code for, so we wouldn't need this here I think

Comment (norhh) [reply mythril/ether/ethcontract.py]: But SolidityContract takes the disassembly from ETHContract right?

Comment (JoranHonig) [reply mythril/ether/ethcontract.py]: Yes, a solidity contract has creation code and is an ETHContract. The ethcontract baseclass takes case of the runtime disassembly. 
However the ETHContract is not a solidity contract and (generally) does not have creation code. The current implementation does have it sometimes, but by default it is ""

I do see that ETHContract already stored the creation_code, so your extension is not illogical.
Lets keep it like this for now

**Comment by norhh** [comment]
@JoranHonig  any additional changes required here?

File: mythril/ether/ethcontract.py (hunk 1 above)
Comment (JoranHonig) [reply mythril/ether/ethcontract.py]: shouldn't it be self.creation_disassembly?

Comment (norhh) [reply mythril/ether/ethcontract.py]: Oops I thought I changed it

Comment (norhh) [reply mythril/ether/ethcontract.py]: Wait! It’s changed right?

**Comment by b-mueller** [comment]
@JoranHonig I think this can be merged?

**Pull request merged**